app/src/test/java/com/codesoom/assignment/application/TaskServiceTest.java

```diff
@@ -0,0 +1,76 @@
+package com.codesoom.assignment.application;
+
+import com.codesoom.assignment.TaskNotFoundException;
+import com.codesoom.assignment.models.Task;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+class TaskServiceTest {
+
+	TaskService taskService;
+
+	@BeforeEach
+	void setUp() {
+		taskService = new TaskService();
+
+		Task task = Task.title("NEW TASK");
+		taskService.createTask(task);
+	}
+
+	@Test
+	void getTasks() {
+		assertThat(taskService.getTasks()).hasSize(1);
+	}
+
+	@Test
+	void getTaskWithValidId() {
+		long id = 1L;
+
+		assertThat(taskService.getTask(id)).isNotNull();
+	}
+
+	@Test
+	void getTaskWithInvalidId() {
+		long id = 99L;
+
+		assertThatThrownBy(() -> taskService.getTask(id)).isInstanceOf(TaskNotFoundException.class);
+	}
```

Comment:
getTask가 주어진 상황(할 일이 존재하는 경우와 존재하지 않는 경우)에 따라 달라지네요. 이런 경우 BDD 테스트 스타일로 테스트를 적으면 주어진 상황을 명시적으로 드러낼 수 있습니다. 예를들어서

```suggestion
@Nested
class Describe_getTask {
    long id = 1L;
    
    @Nested
    class Context_when_task_exsists {
        @BeforeEach
        void setUp() {
            createTaskWithId(id);
        }

	@Test
	void it_returns_task() {
		assertThat(taskService.getTask(id)).isNotNull();
	}
    }

    @Nested
    class Context_when_task_dont_exist {
        @BeforeEach
        void setUp() {
            deleteTaskWithId(1);
        }

	@Test
	void it_returns_task() {
		assertThatThrownBy(() -> taskService.getTask(id)).isInstanceOf(TaskNotFoundException.class);
	}
    }
```

그래서 Describe에는 테스트 대상을
Context에는 주어진 상황을 (그래서 with, when, without 으로만 시작해야 합니다.)

자세한 내용은 종립님 블로그를 한 번 참고해 보세요.

See also
  - https://johngrib.github.io/wiki/junit5-nested/

Reply:
아하.. 생소한 패턴인데, 익숙해지면 상황에 따라 각각 다른 결과를 테스트하기 좋을 것 같습니다.